### PR TITLE
Add file/line info to 'trying next' and vconstructors

### PR DESCRIPTION
**Pull request by ChrisJefferson**
Previously, line/file info was added to trace methods.

This info wasn't in 'trying next' lines, or constructors. Now it is.

First I show some new output, then the old output below. Notice more lines now have file:line info.

The remaining missing lines are:

* system getter : There is no line number :)
* =: for `infinity` and cyclotomic : This is just set to `ReturnFalse`, which is a system call. In general, these line numbers are the actual functions, not the location of the `InstallMethod`, which some users might want, but we can't sensibly store anyway.

I'd like to put this patch into 4.8 as well, but I'm happy if we decide it shouldn't be.

```
gap> TraceAllMethods();;
gap> Union(Rationals, Integers);;
#I  Size: system getter
#I  =: for `infinity' and cyclotomic
#I  Size: system getter
#I  =: for `infinity' and cyclotomic
#I  Union2: for two times the same ring (ResClasses) at /Users/caj/reps/gap/gap/pkg/resclasses/lib/resclass.gi:1716
#I  =: for certain cyclotomic semirings at /Users/caj/reps/gap/gap/lib/cyclotom.gi:2020
#I  CollectionsFamily: for a family at /Users/caj/reps/gap/gap/lib/coll.gi:22
#I  Setter(ElementsFamily): method to inherit `Characteristic' to collections families at /Users/caj/reps/gap/gap/lib/arith.gi:604
#I Trying next: Setter(ElementsFamily): system setter
#I  Setter(CollectionsFamily): system setter
#I  PositionProperty: for dense list and function at /Users/caj/reps/gap/gap/lib/list.gi:1546
#I  PositionProperty: for dense list and function at /Users/caj/reps/gap/gap/lib/list.gi:1546
#I Trying next: Union2: for two collections at /Users/caj/reps/gap/gap/lib/coll.gi:2554
#I Trying next: Union2: for two classes/lists/collections at /Users/caj/reps/gap/gap/pkg/crisp/lib/classes.gi:695
#I  InfoLevel at /Users/caj/reps/gap/gap/lib/info.gi:281
#I  InfoLevel at /Users/caj/reps/gap/gap/lib/info.gi:281
gap> NewFloat(FLOAT_DEFAULT_REP, 1.234);;
#I  NewFloat at /Users/caj/reps/gap/gap/lib/float.gi:124
gap>
```

```
gap> TraceAllMethods();;
gap> Union(Rationals, Integers);;
#I  Size: system getter
#I  =: for `infinity' and cyclotomic
#I  Size: system getter
#I  =: for `infinity' and cyclotomic
#I  Union2: for two times the same ring (ResClasses) at /Users/caj/reps/gap/gap-clean/pkg/resclasses/lib/resclass.gi:1281
#I  =: for certain cyclotomic semirings at /Users/caj/reps/gap/gap-clean/lib/cyclotom.gi:2020
#I  CollectionsFamily: for a family at /Users/caj/reps/gap/gap-clean/lib/coll.gi:22
#I  Setter(ElementsFamily): method to inherit `Characteristic' to collections families at /Users/caj/reps/gap/gap-clean/lib/arith.gi:604
#I  trying next: Setter(ElementsFamily): system setter
#I  Setter(CollectionsFamily): system setter
#I  PositionProperty: for dense list and function at /Users/caj/reps/gap/gap-clean/lib/list.gi:1546
#I  PositionProperty: for dense list and function at /Users/caj/reps/gap/gap-clean/lib/list.gi:1546
#I  trying next: Union2: for two collections
#I  trying next: Union2: for two classes/lists/collections
#I  InfoLevel at /Users/caj/reps/gap/gap-clean/lib/info.gi:281
#I  InfoLevel at /Users/caj/reps/gap/gap-clean/lib/info.gi:281
gap> NewFloat(FLOAT_DEFAULT_REP, 1.234);;
#I  NewFloat
```

### CODE REVIEW

File: lib/methsel.g [hunk 1]
```diff
@@ -213,7 +213,15 @@ end;
...
-
+NEXT_VMETHOD_PRINT_INFO := function ( methods, i, arity)
+    Print("#I Trying next: ", methods[(arity+4)*(i-1)+(arity+4)]);
+    if FILENAME_FUNC(methods[(arity+4)*(i-1)+(arity+2)]) <> fail then
+        Print(" at ",
+              FILENAME_FUNC(methods[(arity+4)*(i-1)+(arity+2)]), ":",
+              STARTLINE_FUNC(methods[(arity+4)*(i-1)+(arity+2)]));
+    fi;
+    Print("\n");
+end;
```
Comment (fingolfin) [inline lib/methsel.g]: Reading `methods[(arity+4)*(i-1)+(arity+2)]` three times in a row like this makes my eyes bleed :-P. Could you perhaps put it into a local variable? That would make the code easier to understand, too.

**Comment by fingolfin** [comment]
Seems like a good idea (with one minor nitpick).

**Comment by alex-konovalov** [comment]
I don't mind this going into stable-4.8, as stable already has many places where line numbers are reported (after introducing a variable for `(arity+4)*(i-1)+(arity+2)`).

**Comment by ChrisJefferson** [comment]
Moved to #655 (as github doesn't let me change what branch a pull request applies to)

**Pull request closed without merging**